test/metrics/prom-metrics-linter/Dockerfile

```
@@ -1,13 +1,17 @@
-FROM golang:1.20 as build
+FROM golang:1.20 AS build
```

Please add `--platform` , to make sure the builder is running with the host arch. Not doing that may fail or be very slow, because the caller uses `--platform` of the target arch.
```suggestion
FROM --platform=${BUILDPLATFORM} golang:1.20 AS build
```

---

Okay, updated.
I haven't encountered any failures without this, but to be on the safe side, we should add it.